core/src/mindustry/logic/LExecutor.java

```diff
@@ -1113,7 +1109,9 @@ public void run(LExecutor exec){
             if(exec.building(target) instanceof MessageBuild d && (d.team == exec.team || exec.privileged)){
 
                 d.message.setLength(0);
-                d.message.append(exec.textBuffer, 0, Math.min(exec.textBuffer.length(), maxTextBuffer));
+                // Backslash is an escape character for format instruction placeholders
+                String result = exec.textBuffer.toString().replace("\\@", "@");
```

Comment:
This will allocate every single print flush. Is there any particular good reason to combine non-formatted "@" with format instructions? (Do we really need escaping?)